src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -257,6 +257,7 @@ namespace AppInstaller::CLI::Workflow
     {
         bool isUpdate = WI_IsFlagSet(context.GetFlags(), Execution::ContextFlag::InstallerExecutionUseUpdate);
         UpdateBehaviorEnum updateBehavior = context.Get<Execution::Data::Installer>().value().UpdateBehavior;
+        bool doUninstallPrevious = isUpdate && (updateBehavior == UpdateBehaviorEnum::UninstallPrevious || context.Args.Contains(Execution::Args::Type::UninstallPrevious));

```

Comment:
This may have impact when dependencies are enabled, but I'm not blocking it since it's in experimental

Reply:
I'm not sure I understand how it would have an impact when dependencies are enabled, but since both are experimental I agree that it's not a blocker. I'll be happy to revisit the argument implementation if necessary